codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddUserAgentDependency.java

```diff
@@ -57,6 +57,9 @@ public void addConfigInterfaceFields(
         writer.writeDocs("The provider populating default tracking information to be sent with `user-agent`, "
                 + "`x-amz-user-agent` header\n@internal");
         writer.write("defaultUserAgentProvider?: Provider<__UserAgent>;\n");
+        writer.writeDocs("The application ID used to identify the application.");
+        writer.write("userAgentAppId?: string | "
```

Comment:
this is already defined in UserAgentInputConfig, and with a different type

it doesn't need to be written inline here

Reply:
regardless of whether this remains, this PR needs full codegen